crates/solver/src/settlement.rs

```diff
@@ -299,6 +318,77 @@ impl Settlement {
+        max_settlement_price_deviation: &Ratio<BigInt>,
+        tokens_to_satisfy_price_test: &TokensForPriceCheck,
+    ) -> bool {
+        if matches!(tokens_to_satisfy_price_test, TokensForPriceCheck::Tokens(token_list) if token_list.is_empty())
+        {
+            return true;
+        }
+        // The following check is quadratic in run-time, although a similar check with linear run-time would also be possible.
+        // For the linear implementation, one would have to find a unique scaling factor that scales the external prices into
+        // the settlement prices. Though, this scaling factor is not easy to define, if reference tokens like ETH are missing.
+        // Since the checks would heavily depend on this scaling factor, and its
+        // derivation is non-trivial, we decided to go for the implementation with quadratic run time. Settlements
+        // will not have enough tokens, such that the run-time is important.
+        self
+            .clearing_prices()
+            .iter()
+            .combinations(2)
+            .all(|clearing_price_vector_combination| {
+                let (sell_token, sell_price) = clearing_price_vector_combination[0];
+                let clearing_price_sell_token = sell_price.to_big_rational();
+                let (buy_token, buy_price) = clearing_price_vector_combination[1];
+                let clearing_price_buy_token = buy_price.to_big_rational();
+
+                if matches!(tokens_to_satisfy_price_test, TokensForPriceCheck::Tokens(token_list) if (!token_list.contains(sell_token)) || !token_list.contains(buy_token))
+                {
+                    return true;
+                }
+                let external_price_sell_token = match external_prices.price(sell_token) {
+                    Some(price) => price,
+                    None => return true,
+                };
+                let external_price_buy_token = match external_prices.price(buy_token) {
+                    Some(price) => price,
+                    None => return true,
+                };
+                // Condition to check: Deviation of clearing prices is bigger than max_settlement_price deviation
+                //
+                // |clearing_price_sell_token / clearing_price_buy_token - external_price_sell_token/external_price_buy_token)|
+                // |----------------------------------------------------------------------------------------------------------|
+                // |                     clearing_price_sell_token / clearing_price_buy_token                                  |
+                // is bigger than:
+                // max_settlement_price_deviation
+                //
+                // This is equal to: |clearing_price_sell_token*external_price_buy_token-external_price_sell_token*clearing_price_buy_token|>
+                // max_settlement_price_deviation*clearing_price_buy_token*external_price_buy_token *clearing_price_sell_token
```

Comment:
It's a shame the names here are so long. Really make the equations hard to read. 
```suggestion
                // This is equal to: |clearing_price_sell_token * external_price_buy_token - external_price_sell_token * clearing_price_buy_token|>
                // max_settlement_price_deviation * clearing_price_buy_token * external_price_buy_token * clearing_price_sell_token
```